### PR TITLE
Sets default configFolder to config/default

### PR DESCRIPTION
**Pull request by h0tbird**
**What this PR does / why we need it**:
Fixes the error below:
```
 cluster-api git:(master) ./cmd/clusterctl/hack/local-overrides.py
clusterctl local overrides generated from local repositories for the cluster-api, kubeadm-bootstrap, kubeadm-control-plane, aws providers.
in order to use them, please run:

clusterctl init --core cluster-api:v0.3.0 --bootstrap kubeadm-bootstrap:v0.3.0 --control-plane kubeadm-control-plane:v0.3.0 --infrastructure aws:v0.5.0

 cluster-api git:(master) cat ~/.cluster-api/overrides/aws/v0.5.0/infrastructure-components.yaml
Error: unable to find one of 'kustomization.yaml', 'kustomization.yml' or 'Kustomization' in directory '/Users/mvillacorta/git/newrelic/cluster-api-provider-aws/config'
```

If this PR is merged then also consider this CAPA PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1571

### CODE REVIEW

**Comment by h0tbird** [comment]
Alternatively another solution might be adding the `/config/kustomization.yaml` to CAPA and all the other infrastructure providers?

File: cmd/clusterctl/hack/local-overrides.py [hunk 1]
```diff
@@ -140,7 +141,7 @@ def create_local_overrides():
         assert p is not None, 'invalid configuration: please specify the configuration for the {} provider'.format(provider)
 
         repo = p.get('repo', '.')
-        config_folder = p.get('configFolder', 'config')
+        config_folder = p.get('configFolder', 'config/default')
```
Comment (fabriziopandini) [inline cmd/clusterctl/hack/local-overrides.py]: After the changes introduced by https://github.com/kubernetes-sigs/cluster-api/pull/2279 the root folder for generating manifest is `config` (not anymore `config/default`), so this value should not be changed.

Comment (h0tbird) [reply cmd/clusterctl/hack/local-overrides.py]: Thank you @fabriziopandini I will try to fix this in CAPA then.

**Comment by h0tbird** [comment]
Thank you! Closing.

**Pull request closed without merging**